### PR TITLE
Verify no data is written to vol dir after tests are run.

### PR DESCRIPTION
**Pull request by piyush-nimbalkar**

### CODE REVIEW

File: drivers/scheduler/k8s/k8s.go [hunk 1]
```diff
@@ -458,17 +459,21 @@ func (k *k8s) waitForCleanup(ctx *scheduler.Context, podList []v1.Pod) error {
...
+		MaxDepth: 1,
+	}
 
 	for _, n := range k.GetNodes() {
 		if n.Type == node.TypeWorker {
-			if exists, err := driver.CheckIfPathExists(podVolDir, n, options); err != nil {
+			if volDir, err := driver.FindFiles(podVolDir, n, options); err != nil {
 				return err
-			} else if exists {
+			} else if strings.TrimSpace(volDir) != "" {
```
Comment (harsh-px) [inline drivers/scheduler/k8s/k8s.go]: need to include `volDir` in the error message so it propagates up explicitly in the logs.

File: drivers/volume/portworx/schedops/errors.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package schedops
+
+import (
+	"fmt"
+	"strings"
+)
+
+// ErrFailedToCleanupVolume error type for orphan pods or unclean vol directories
+type ErrFailedToCleanupVolume struct {
```
Comment (harsh-px) [inline drivers/volume/portworx/schedops/errors.go]: :+1:

File: drivers/volume/portworx/schedops/k8s-schedops.go [hunk 3]
```diff
@@ -30,6 +36,96 @@ func (k *k8sSchedOps) EnableOnNode(n node.Node) error {
...
+			}
+			if found {
+				continue
+			}
+			orphanPods = append(orphanPods, podUID)
+
+			// check if there are files under portworx volume
+			n := nodeMap[nodeName]
+			findFileOpts := node.FindOpts{
+				ConnectionOpts: connOpts,
```
Comment (harsh-px) [inline drivers/volume/portworx/schedops/k8s-schedops.go]: Can you add an comment showing example why depth of 2 is selected here with the path regex *portworx-volume?

File: drivers/volume/portworx/schedops/k8s-schedops.go [hunk 4]
```diff
@@ -30,6 +36,96 @@ func (k *k8sSchedOps) EnableOnNode(n node.Node) error {
...
+
+func separateFilePaths(volDirList string) []string {
+	trimmedList := strings.TrimSpace(volDirList)
+	if trimmedList == "" {
+		return []string{}
+	}
+	return strings.Split(trimmedList, "\n")
+}
+
+func extractPodUID(volDirPath string) string {
```
Comment (harsh-px) [inline drivers/volume/portworx/schedops/k8s-schedops.go]: Better to use a regex than just depending on the 5th item being pod uid.

File: drivers/scheduler/k8s/k8s.go (hunk 1 above)
Comment (piyush-nimbalkar) [reply drivers/scheduler/k8s/k8s.go]: the volDir is /var/lib/kubelet/pods/<pod_uid>/volumes/*
, where the only imp thing is the pod UID

Comment (harsh-px) [reply drivers/scheduler/k8s/k8s.go]: Ok. I thought it has the volume name too. If not, no change is needed.

File: drivers/volume/portworx/schedops/k8s-schedops.go (hunk 4 above)
Comment (piyush-nimbalkar) [reply drivers/volume/portworx/schedops/k8s-schedops.go]: Done

File: drivers/volume/portworx/schedops/k8s-schedops.go (hunk 3 above)
Comment (piyush-nimbalkar) [reply drivers/volume/portworx/schedops/k8s-schedops.go]: Done

File: drivers/node/node.go [hunk 5]
```diff
@@ -66,8 +74,8 @@ type Driver interface {
 	// ShutdownNode shuts down the given node
 	ShutdownNode(node Node, options ShutdownNodeOpts) error
 
-	// CheckIfPathExists checks whether the given path is present in the node
-	CheckIfPathExists(path string, node Node, options ConnectionOpts) (bool, error)
+	// FindFiles finds and returns the files for the given path regex and the node
+	FindFiles(path string, node Node, options FindOpts) (string, error)
```
Comment (harsh-px) [inline drivers/node/node.go]: Add a default implementation for aws node driver. Without that, the aws tests will start failing with FindFiles() method not supported error.

Comment (piyush-nimbalkar) [reply drivers/node/node.go]: oops missed it.. adding it.

**Pull request merged**